design-doc.md

```diff
@@ -0,0 +1,150 @@
+# Design Document: Adding Support for Next Link Paging with Re-injected Query Parameters
+
+## Overview
+
+Currently, we treat the `nextLink` in paging as opaque and does not inject query parameters that were part of the initial request when following the next link. This proposal aims to enhance our `nextLink` handling by adding capability to reinject query parameters into next requests for paging operations.
```

Comment:
Are we limiting this to reinjecting only query params or should we include request headers too?

Reply:
i believe we should limit this to only query params. Are there instances / is this something we want to consider? I'll bring this up in the meeting as a question to answer as well